sickbeard/providers/kat.py

```diff
@@ -66,81 +76,92 @@ def imageName(self):
+                logger.log(u"Search string: %s" % unicode(self.search_params), logger.DEBUG)
 
-                if mode != 'RSS':
-                    searchURL = self.url + 'usearch/%s/?field=seeders&sorder=desc&rss=1' % urllib.quote_plus(search_string)
-                else:
-                    searchURL = self.url + 'tv/?field=time_add&sorder=desc&rss=1'
+                try:
+                    data = self.getURL(self.urls[('search', 'rss')[mode == 'RSS']], params=self.search_params)
+                    if not data:
+                        continue
 
-                logger.log(u"Search string: " + searchURL, logger.DEBUG)
+                    entries = xmltodict.parse(data)
+                    if not all([entries, 'rss' in entries, 'channel' in entries['rss'], 'item' in entries['rss']['channel']]):
+                        continue
 
-                try:
-                    entries = self.cache.getRSSFeed(searchURL)['entries']
-                    for item in entries or []:
+                    for item in entries['rss']['channel']['item']:
                         try:
-                            link = item['link']
-                            id = item['guid']
                             title = item['title']
-                            url = item['torrent_magneturi']
-                            verified = bool(int(item['torrent_verified']) or 0)
-                            seeders = int(item['torrent_seeds'])
-                            leechers = int(item['torrent_peers'])
-                            size = int(item['torrent_contentlength'])
+
+                            # Use the torcache link kat provides,
+                            # unless it is not torcache or we are not using blackhole
+                            # because we want to use magnets if connecting direct to client
+                            # so that proxies work.
+                            url = item['enclosure']['@url']
+                            if sickbeard.TORRENT_METHOD != "blackhole" or 'torcache' not in url:
+                                url = item['torrent:magnetURI']
+
+                            seeders = int(item['torrent:seeds'])
+                            leechers = int(item['torrent:peers'])
+                            verified = bool(int(item['torrent:verified']) or 0)
+                            size = int(item['torrent:contentLength'])
+
+                            info_hash = item['torrent:infoHash']
+                            #link = item['link']
+
                         except (AttributeError, TypeError, KeyError):
                             continue
 
-                        if mode != 'RSS' and (seeders < self.minseed or leechers < self.minleech):
+                        # Dont let RSS add items with no seeders either -.-
+                        if not seeders or seeders < self.minseed or leechers < self.minleech:
```

Comment:
It shouldnt be added to RSS, because the user has said he doesnt want torrents with less than 10 seeders. With that other fix RSS adds torrents with 0 seeders.